### PR TITLE
Xtest keyboard simulation for Xlib

### PR DESCRIPTION
**Pull request by RobertBColton**
This adds the requested `keyboard_key_press`/`keyboard_key_release` from #1054 for Xlib as an extension using the XTEST Extension Protocol.

Edit: Here's proof from Hugar that the functions work to simulate pressing the 'A' key in the terminal:
Keyboard Simulation of 'A' key in the Terminal

### CODE REVIEW

File: ENIGMAsystem/SHELL/Platforms/xlib/Makefile [hunk 1]
```diff
@@ -1,2 +1,2 @@
 SOURCES += $(wildcard Platforms/xlib/*.cpp) $(wildcard Platforms/General/POSIX/*.cpp)
-override LDLIBS += -lz -lpthread -lX11
+override LDLIBS += -lz -lpthread -lX11 -lXtst
```
Comment (RemoveRusky) [inline ENIGMAsystem/SHELL/Platforms/xlib/Makefile]: move dis shit

File: ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <X11/extensions/XTest.h>
```
Comment (RemoveRusky) [inline ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp]: move dis shit as well

File: ENIGMAsystem/SHELL/Universal_System/Extensions/XTEST/About.ey [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+---
+
+Name: XTEST
+Identifier: XTEST
+Author: Robert Colton
+Description: Adds Xlib operating system-level keyboard simulation using XTEST Extension Protocol
+Default: false
+Icon: xtest.png
+
+Depends: None
```
Comment (RemoveRusky) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/XTEST/About.ey]: depends platform xlib dont it?

File: ENIGMAsystem/SHELL/Universal_System/Extensions/XTEST/xtest.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#include "include.h"
+
+#include "Platforms/xlib/XLIBwindow.h"
+
+namespace enigma {
+  extern unsigned short keyrmap[256];
```
Comment (RemoveRusky) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/XTEST/xtest.cpp]: find header this shits in. if it aint in one put it somewhere.

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Universal_System/Extensions/XTEST/xtest.cpp]: Done: 2a58d7a0fdb49fe8d8da94e26f018017dde971e8

But how does this affect Windows cross-compilation? Would the extension be available?

**Comment by RobertBColton** [comment]
Closing as stale in favor of #1463.

**Pull request closed without merging**